### PR TITLE
pool

### PR DESCRIPTION
**Pull request by Kranium2002**
Issue id: #1548 

Added:

- Function for tensorflow frontend nn.pool. Also, created it in layers.py with docstring.
- Tests for the function nn.pool

### CODE REVIEW

**Comment by Kranium2002** [comment]
@kurshakuz Could you help me with this pr?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py [hunk 1]
```diff
@@ -1451,3 +1451,52 @@ def test_tensorflow_avg_pool3d(
...
+    data_format = data_formats[dims - 1]
+    x = draw(
+        helpers.arrays_for_pooling(
+            min_dims=dims + 2, max_dims=dims + 2, min_side=1, max_side=4
+        )
+    )
+
+    dilation = draw(st.sampled_from([None, [1] * dims, [2] * dims]))
+    window_size = [
+        draw(st.integers(min_value=1, max_value=x[0].shape[i + 1]) | st.just(1)) for _ in range(dims)
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1469]:
```
E   AttributeError: 'list' object has no attribute 'shape'
```

Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1469]: ```i``` here is not defined anywhere

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Hi @Kranium2002! Thank you for your contribution!

I have tested your implementation locally and it is currently failing. I have attached a review why it is the case.

Please resolve that issues and test it like this:
```
pytest ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py::test_tensorflow_pool
```

Please let me know once you get that passing. Thanks!

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py (hunk 1 above)
Comment (Kranium2002) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py]: Hi, I'm facing some errors while writing the pool function. I've given the full description of my problem in the commit summary, if you require anymore information please contact me.

Thank you for helping me

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Pull request closed without merging**